lib/adal.js

```diff
@@ -494,6 +495,113 @@ AuthenticationContext.prototype.acquireToken = function (resource, callback) {
     }
 };
 
+/**
+ * Acquires token (interactive flow using a redirect) by sending request to AAD to obtain a new token. In this case the callback passed in the Authentication
+ * request constructor will be called.
+ * @param {string}   resource  ResourceUri identifying the target resource
+ * @param {string}   extraQueryParameters  extraQueryParameters to add to the authentication request
+ */
+AuthenticationContext.prototype.acquireTokenRedirect = function (resource, extraQueryParameters, claims) {
+    if (this._isEmpty(resource)) {
+      this.warn('resource is required');
+      callback('resource is required', null, 'resource is required');
+      return;
+    }
+  
+    var callback = this.callback;
+  
+    if (!this._user) {
+      this.warn('User login is required');
+      callback('User login is required', null, 'login required');
+      return;
+    }
+  
+    if (this._acquireTokenInProgress) {
+      this.warn("Acquire token interactive is already in progress")
+      callback("Acquire token interactive is already in progress", null, "Acquire token interactive is already in progress");
+      return;
+    }
+  
+    var expectedState = this._guid() + '|' + resource;
+    this.config.state = expectedState;
+    this.verbose('Renew token Expected state: ' + expectedState);
+  
+    // remove the existing prompt=... query parameter and add prompt=select_account
+    var urlNavigate = this._urlRemoveQueryStringParameter(this._getNavigateUrl('token', resource), 'prompt');
+    urlNavigate = urlNavigate + '&prompt=select_account';
+    if (extraQueryParameters) {
+      urlNavigate += extraQueryParameters;
+    }
+  
+    if (claims && (urlNavigate.indexOf("&claims") === -1)) {
+      urlNavigate += '&claims=' + encodeURIComponent(claims);
+    }
+    else if (claims && (urlNavigate.indexOf("&claims") !== -1)) {
+      throw new Error('Claims cannot be passed as an extraQueryParameter');
+    }
+  
+    urlNavigate = this._addHintParameters(urlNavigate);
+    this._acquireTokenInProgress = true;
```

Comment:
do you need to set _acquireTokenInProgress  back to false somewhere?

Reply:
After calling promptUser, it will redirect to login.microsoftonline.com's page and once user login again, we will have fresh code.